mmv1/templates/terraform/examples/discoveryengine_schema_basic.tf.erb

```diff
@@ -0,0 +1,17 @@
+resource "google_discovery_engine_schema" "basic" {
+  location                    = google_discovery_engine_data_store.basic.location
+  data_store_id               = google_discovery_engine_data_store.basic.data_store_id
+  schema_id                   = "<%= ctx[:vars]['schema_id'] %>"
+  json_schema                 = "{\"$schema\":\"https://json-schema.org/draft/2020-12/schema\",\"date_detection\":true,\"type\":\"object\",\"geolocation_detection\":true}"
```

Comment:
```suggestion
  json_schema                 = "{\"schema\":\"https://json-schema.org/draft/2020-12/schema\",\"date_detection\":true,\"type\":\"object\",\"geolocation_detection\":true}"
```

Reply:
Yes, `$` is needed. 

I don't think `$` is the root cause, because 
1. actually this is a valid JSON string.
2. I tested locally by passing `"{\"date_detection\":true,\"type\":\"object\",\"geolocation_detection\":true}"` (no `$schema`), and the same error happened.

I feel it's related to this line `custom_expand: 'templates/terraform/custom_expand/json_schema.erb'`, though I cannot ensure how it impacts the input JSON string. 

From the error message `"Starting an object on a scalar field"`, I feel the error is related to JSON object (with `{` and `}` ) and scalar value (integer, double, string, etc). 
`json_string` is a of string type, which should accept any valid JSON string (including JSON object), but now it seems it can only accept scalar value.

Reply:
`$` is not the root cause here.

I checked the GET API response. The keys inside json are shifted randomly in the response.

Post request
```
{
 "jsonSchema": "{\"$schema\": \"https://json-schema.org/draft/2020-12/schema\", \"type\": \"object\", \"date_detection\": true, \"geolocation_detection\": true}"
}
``` 
Get response
```
{
  "name": "projects/1067888929963/locations/global/collections/default_collection/dataStores/tf-test-data-store-idt4fq1oc2xn/schemas/tf-test-schema-idt4fq1oc2xn",
  "jsonSchema": "{\"$schema\":\"https://json-schema.org/draft/2020-12/schema\",\"date_detection\":true,\"type\":\"object\",\"geolocation_detection\":true}"
}
```

To fix the issue, a new `custom_flatten` file is needed. Inside the new file, we need to normalize the json.

`custom_expand` is not needed, as API needs a json string.

Reply:
Thanks a lot for helping with this issue, your idea worked and the create test was fixed. 

In addition to (1) removing `custom_expand` and (2) defining new `custom_flatten` file, I also (3) added line `primary_resource_id: 'basic'` back (https://github.com/GoogleCloudPlatform/magic-modules/pull/11018#discussion_r1709968807). 

This line is needed to generate [ResourceName](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-11018-old..auto-pr-11018#diff-15348380ff1c87be4b726ce16e88de6336ddcfd5fde294103444ca8db92e78baR49).
If not adding `primary_resource_id: 'basic'`, the value of `ResourceName` will be 
`ResourceName: "google_discovery_engine_schema.",` an error `vcr_utils.go:152: Resource specified by ResourceName couldn't be found: google_discovery_engine_schema.` will happen in acceptance test.